### ISSUE TITLE
Group name is limited to rust identifiers since v0.6

### ISSUE THREAD

**Issue by nickelc** [serenity-rs/serenity#622]
```
error: proc macro panicked
  --> src/main.rs:55:1
   |
55 | / group!({
56 | |     name: "Foo Bar",
57 | |     options: {
58 | |         // Sets multiple prefixes for a group.
...  |
68 | |     commands: [cat, dog],
69 | | });
   | |___^
   |
   = help: message: `"Foo Bar"` is not a valid identifier
```

**Comment by arqunis** [comment]
This is on purpose as whatever inside is used to compose the identifier of the group's generated static instances. For example:
```rust
group!({
    name: "foo",
    commands: [foo, bar, baz],
});

// => `FOO_GROUP_OPTIONS`
// => `FOO_GROUP`
```

Additionally, group names aren't even considered when parsing a message for a possible command, so there's no use for them in that field anyway. 

However, they do affect the display of the help page about them. Something you may want to replace with a more appropriate name for your audience, which you'll be allowed to modify with a `help_name` option I'm going to introduce soon.

**Comment by nickelc** [comment]
It's pretty much the same.
```
..serenity/examples/05_command_framework (git)-[current] % cargo c
   Compiling serenity v0.6.0 (/serenity)
   Compiling command_attr v0.1.1 (/serenity/command_attr)
    Checking 05_command_framework v0.1.0 (/serenity/examples/05_command_framework)
error: proc macro panicked
  --> src/main.rs:71:1
   |
71 | / group!({
72 | |     name: "math",
73 | |     help_name: "math.io",
74 | |     options: {
...  |
80 | |     commands: [multiply],
81 | | });
   | |___^
   |
   = help: message: `"math.io"` is not a valid identifier

error: aborting due to previous error

error: Could not compile `05_command_framework`.
```

**Comment by arqunis** [comment]
Oh, sorry, I mindlessly copied the parsing code for `name`. I'll deploy a fix right now.

**Comment by arqunis** [comment]
Uh, whoops. I meant to give that commit the #596 issue. But it doesn't matter for this anyway, 
This should be resolved now in [c:1ba2279](https://github.com/serenity-rs/serenity/commit/1ba22793c6bbf8f6f36ae52a3f37191256e78dd2).
